lib/pure/algorithm.nim

```diff
@@ -561,36 +702,54 @@ proc rotateLeft*[T](arg: var openarray[T]; slice: HSlice[int, int]; dist: int):
   ##   The distance in amount of elements that the data should be rotated.
   ##   Can be negative, can be any number.
   ##
-  ## .. code-block:: nim
-  ##
-  ##   var list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
-  ##   list.rotateLeft(1 .. 8, 3)
-  ##   doAssert list == [0, 4, 5, 6, 7, 8, 1, 2, 3, 9, 10]
+  ## **See also:**
+  ## * `rotatedLeft proc<#rotatedLeft,openArray[T],HSlice[int,int],int>`_
+  ## * `rotatedLeft proc<#rotatedLeft,openArray[T],int>`_
+  runnableExamples:
+    var list = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    list.rotateLeft(1 .. 8, 3)
+    assert list == [0, 4, 5, 6, 7, 8, 1, 2, 3, 9, 10]
   let sliceLen = slice.b + 1 - slice.a
   let distLeft = ((dist mod sliceLen) + sliceLen) mod sliceLen
   arg.rotateInternal(slice.a, slice.a+distLeft, slice.b + 1)
 
 proc rotateLeft*[T](arg: var openarray[T]; dist: int): int {.discardable.} =
   ## Default arguments for slice, so that this procedure operates on the entire
   ## ``arg``, and not just on a part of it.
+  ##
+  ## **See also:**
+  ## * `rotatedLeft proc<#rotatedLeft,openArray[T],HSlice[int,int],int>`_
+  ## * `rotatedLeft proc<#rotatedLeft,openArray[T],int>`_
   runnableExamples:
-      var a = [1, 2, 3, 4, 5]
-      a.rotateLeft(2)
-      doAssert a == [3, 4, 5, 1, 2]
+    var a = [1, 2, 3, 4, 5]
```

Comment:
As above, an example with multiple rotations (showing the result after each rotation) would be more interesting/informative.

Reply:
Also, in one of these rotating procs, show an example where you use negative `dist`.